### PR TITLE
Adds adoptChildren method

### PR DESCRIPTION
**Pull request by kimpepper**

### CODE REVIEW

File: src/Storage/DbalNestedSet.php [hunk 1]
```diff
@@ -349,17 +355,37 @@ public function moveSubTreeAfter(Node $target, Node $node) {
...
+   * @param int $newDepth
+   *   Depth of new position.
+   *
+   * @throws \Exception
+   *   If a transaction error occurs.
+   */
+  protected function moveMultipleSubTreesToPosition($newLeftPosition, array $nodes, $newDepth) {
     try {
+
+      $firstNode = $nodes[0];
```
Comment (larowlan) [inline src/Storage/DbalNestedSet.php]: ```reset($nodes)```?

File: src/Storage/DbalNestedSet.php [hunk 2]
```diff
@@ -349,17 +355,37 @@ public function moveSubTreeAfter(Node $target, Node $node) {
...
+      $lastNode = $nodes[count($nodes) - 1];
```
Comment (larowlan) [inline src/Storage/DbalNestedSet.php]: ```end($nodes)```?

File: src/Storage/DbalNestedSet.php [hunk 3]
```diff
@@ -349,17 +355,37 @@ public function moveSubTreeAfter(Node $target, Node $node) {
...
+    $this->moveMultipleSubTreesToPosition($newLeftPosition, [$node], $newDepth);
+  }
+
+  /**
+   * Moves multiple subtrees to a new position.
+   *
+   * @param int $newLeftPosition
+   *   The new left position.
+   * @param \PNX\NestedSet\Node[] $nodes
+   *   The node to move.
```
Comment (larowlan) [inline src/Storage/DbalNestedSet.php]: nit: nodes to move

File: src/Storage/DbalNestedSet.php [hunk 4]
```diff
@@ -349,17 +355,37 @@ public function moveSubTreeAfter(Node $target, Node $node) {
...
+  protected function moveMultipleSubTreesToPosition($newLeftPosition, array $nodes, $newDepth) {
     try {
+
+      $firstNode = $nodes[0];
+      $lastNode = $nodes[count($nodes) - 1];
       // Calculate position adjustment variables.
-      $width = $node->getRight() - $node->getLeft() + 1;
-      $distance = $newLeftPosition - $node->getLeft();
-      $tempPos = $node->getLeft();
+      $width = $lastNode->getRight() - $firstNode->getLeft() + 1;
```
Comment (larowlan) [inline src/Storage/DbalNestedSet.php:380]: We only call this internally - but should we be checking that $nodes are consecutive and ordered?

**Review by larowlan — COMMENTED** [review]
slick

File: src/Storage/DbalNestedSet.php (hunk 4 above)
Comment (kimpepper) [reply src/Storage/DbalNestedSet.php:380]: How can we do that?

**Pull request merged**